api/notd/collection_processor.py

```diff
@@ -0,0 +1,62 @@
+import asyncio
+import json
+import logging
+
+from core.exceptions import NotFoundException
+from core.requester import Requester
+from core.requester import ResponseException
+from core.web3.eth_client import EthClientInterface
+
+from notd.model import RetrievedCollection
+
+
+class CollectionDoesNotExist(NotFoundException):
+    pass
+
+class CollectionProcessor:
+    def __init__(self, requester: Requester, ethClient: EthClientInterface):
+        self.requester = requester
+        self.ethClient = ethClient
+        with open('./contracts/IERC721Metadata.json') as contractJsonFile:
+            erc721MetdataContractJson = json.load(contractJsonFile)
+        self.erc721MetdataContractAbi = erc721MetdataContractJson['abi']
+        self.erc721MetdataUriFunctionAbi = [internalAbi for internalAbi in self.erc721MetdataContractAbi if internalAbi['name'] == 'tokenURI'][0]
+        self.erc721MetadataNameFunctionAbi = [internalAbi for internalAbi in self.erc721MetdataContractAbi if internalAbi['name'] == 'name'][0]
+        self.erc721MetadataSymbolFunctionAbi = [internalAbi for internalAbi in self.erc721MetdataContractAbi if internalAbi['name'] == 'symbol'][0]
+
+
+    async def retrieve_collection(self,address) -> RetrievedCollection:
+        response = 0
+        retryCount = 0
+        tokenMetadataNameResponse = await self.ethClient.call_function(toAddress=address, contractAbi=self.erc721MetdataContractAbi, functionAbi=self.erc721MetadataNameFunctionAbi)
+        tokenMetadataSymbolResponse = await self.ethClient.call_function(toAddress=address, contractAbi=self.erc721MetdataContractAbi, functionAbi=self.erc721MetadataSymbolFunctionAbi)
```

Comment:
for this and above, many registries may not actually implement this so we should catch an exception here

Reply:
Okay do we send a log message or just pass?

Reply:
yep print a log message similar to the token processor